src/dali/plugin/input/csv/blockfi.py

```diff
@@ -93,15 +100,34 @@ def load(self) -> List[AbstractTransaction]:
                             fiat_fee="0",
                         )
                     )
-                elif line[self.__TYPE_INDEX] == _CRYPTO_TRANSFER:
+                elif entry_type == _REFERRAL_BONUS:
                     last_withdrawal_fee = None
                     result.append(
-                        IntraTransaction(
+                        InTransaction(
                             plugin=self.__BLOCKFI,
                             unique_id=Keyword.UNKNOWN.value,
                             raw_data=raw_data,
                             timestamp=f"{line[self.__TIMESTAMP_INDEX]} -00:00",
                             asset=line[self.__CURRENCY_INDEX],
+                            exchange=self.__BLOCKFI,
+                            holder=self.account_holder,
+                            transaction_type="Income",
+                            spot_price=Keyword.UNKNOWN.value,
+                            crypto_in=line[self.__AMOUNT_INDEX],
+                            fiat_fee="0",
+                            notes="Referral Bonus"
+                        )
+                    )
+                elif entry_type == _CRYPTO_TRANSFER:
+                    last_withdrawal_fee = None
+                    asset = line[self.__CURRENCY_INDEX]
```

Comment:
This is the only place where asset is assigned to a variable: for consistency with the rest of the code, can we remove the variable and just use `line[self.__CURRENCY_INDEX]` inline? An alternative would be to factor asset at the top of the file into a variable, but then we would have to do the same to all other "factorable" values (e.g. timestamp). I lean toward the first option.